packages/babel-types/test/cloning.js

```diff
@@ -72,4 +72,21 @@ describe("cloneNode", function() {
       node.declarations[0].id.typeAnnotation,
     );
   });
+
+  it("should support deep cloning without loc", function() {
+    const node = t.variableDeclaration("let", [
+      t.variableDeclarator({
+        ...t.identifier("value"),
+        typeAnnotation: t.anyTypeAnnotation(),
+      }),
+    ]);
+    const cloned = t.cloneNode(node, /* deep */ true, /* withoutLoc */ true);
+    expect(cloned.declarations[0].id.typeAnnotation).toEqual(
```

Comment:
We can remove the `typeAnnotation` checks, we're only interested in `loc` for the test.